src/box/applier.cc

```diff
@@ -1378,6 +1378,7 @@ apply_plain_tx(uint32_t replica_id, struct stailq *rows,
 	struct applier_tx_row *item;
 	if (txn == NULL)
 		 return -1;
+	txn->isolation = TXN_ISOLATION_READ_COMMITTED;
```

Comment:
I'd probably use `box_txn_set_isolation` here. Then we'll be safe in case setting txn isolation becomes more complex than just setting a variable someday.

Or even better, to avoid the extra checks `box_txn_set_isolation` performs, you can extract the part which sets the isolation in a separate routine and use it here.
And also use it in `box_txn_set_isolation` after the checks.

Reply:
Not sure about this. `box_txn_set_isolation` is really general: even if we factor out the part that sets the passed transaction's isolation level, it still performs additional checks, e.g., whether the isolation level is default, and whether the transaction needs to be linearized.

@alyapunov RFC

Reply:
I agree with @CuriousGeorgiy, box_txn_set_isolation is a part of API but we do some internal thing here and there's no reason to do additional check.

Perhaps one day we'll need a wrapper like `txn_set_isolation` but it seems to be an overkill today.

Reply:
Besides, we didn't do this this in vinyl: https://github.com/tarantool/tarantool/blob/a24714c55305280838576fa6dec9a9781763e893/src/box/vy_tx.c#L344.

Reply:
@locker what do you think about this?

Reply:
Vinyl inherits the isolation level from `struct txn`:

https://github.com/tarantool/tarantool/blob/02e2b2920e2c99c349bcfc246976a0f3e0c2a199/src/box/vinyl.c#L2433

Still, it'd be nice to have a test that makes sure this patches fixes the issue for both engines.